### PR TITLE
chore(samples): added samples for Regional endpoint

### PR DESCRIPTION
**Pull request by shweta345**
## Description

Fixes #276856509

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: firestore/cloud-client/noxfile_config.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Copyright 2019 Google LLC
```
Comment (Mariatta) [inline firestore/cloud-client/noxfile_config.py:1]:
```suggestion
# Copyright 2023 Google LLC
```

File: firestore/cloud-client/snippets_test.py [hunk 2]
```diff
@@ -330,3 +330,6 @@ def test_list_document_subcollections():
 def test_create_and_build_bundle():
     bundle, buffer = snippets.create_and_build_bundle()
     assert "latest-stories-query" in bundle.named_queries
+
+def test_regional_endpoint():
+    snippets.regional_endpoint()
```
Comment (Mariatta) [inline firestore/cloud-client/snippets_test.py:335]:
```suggestion
    snippets.regional_endpoint()

```

Comment (Mariatta) [reply firestore/cloud-client/snippets_test.py:335]: Suggest adding empty line break at the end of file.

**Comment by shweta345** [comment]
@Mariatta Added the datastore snippets. Though the regional endpoint test passes, other tests are failing: https://paste.googleplex.com/6383708654272512

File: datastore/cloud-client/snippets.py [hunk 3]
```diff
@@ -883,6 +884,21 @@ def property_by_kind_run_query(client):
...
+    # [START datastore_regional_endpoints]
+    ENDPOINT = "https://datastore.googleapis.com"
+    client_options = ClientOptions(api_endpoint=ENDPOINT)
+    client = datastore.Client(client_options=client_options)
+ 
+    query = client.query(kind="Task")
+    results = list(query.fetch())
+    for r in results:
+        print(r)
+    # [End datastore_regional_endpoints]
```
Comment (Mariatta) [inline datastore/cloud-client/snippets.py:897]:
```suggestion
    # [END datastore_regional_endpoints]
```

File: datastore/cloud-client/snippets_test.py [hunk 4]
```diff
@@ -301,3 +301,8 @@ def test_property_by_kind_run_query(self, client):
     @backoff.on_exception(backoff.expo, AssertionError, max_time=240)
     def test_index_merge_queries(self, client):
         snippets.index_merge_queries(client)
+
+    @backoff.on_exception(backoff.expo, AssertionError, max_time=240)
+    def test_regional_endpoint(self, client):
+        client = snippets.regional_endpoint()
+        assert client is not None
```
Comment (Mariatta) [inline datastore/cloud-client/snippets_test.py:308]:
```suggestion
        assert client is not None
 
```

File: firestore/cloud-client/snippets_test.py [hunk 5]
```diff
@@ -330,3 +330,7 @@ def test_list_document_subcollections():
...
+
```
Comment (Mariatta) [inline firestore/cloud-client/snippets_test.py:336]:
```suggestion

```

**Review by Mariatta — APPROVED** [review]

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**